src/Compilers/CSharp/Portable/Binder/PatternSwitchBinder2.cs

```diff
@@ -0,0 +1,234 @@
+            out BoundPatternSwitchLabel defaultLabel,
+            out bool isComplete,
+            out bool someCaseMatches,
+            DiagnosticBag diagnostics)
+        {
+            defaultLabel = null;
+
+            // someCaseMatches will be set to true if some single case label would handle all inputs
+            someCaseMatches = false;
+
+            // Bind match sections
+            var boundPatternSwitchSectionsBuilder = ArrayBuilder<BoundPatternSwitchSection>.GetInstance();
+            SubsumptionDiagnosticBuilder subsumption = new SubsumptionDiagnosticBuilder(ContainingMemberOrLambda, SwitchSyntax, this.Conversions, SwitchGoverningType);
+            foreach (var sectionSyntax in SwitchSyntax.Sections)
+            {
+                var section = BindPatternSwitchSection(sectionSyntax, originalBinder, ref defaultLabel, ref someCaseMatches, subsumption, diagnostics);
+                boundPatternSwitchSectionsBuilder.Add(section);
+            }
+
+            isComplete = defaultLabel != null || subsumption.IsComplete || someCaseMatches;
+            return boundPatternSwitchSectionsBuilder.ToImmutableAndFree();
+        }
+
+        /// <summary>
+        /// Bind the pattern switch section, producing subsumption diagnostics.
+        /// </summary>
+        /// <param name="node"/>
+        /// <param name="originalBinder"/>
+        /// <param name="defaultLabel">If a default label is found in this section, assigned that label</param>
+        /// <param name="someCaseMatches">If a case is found that would always match the input, set to true</param>
+        /// <param name="subsumption">A helper class that uses a decision tree to produce subsumption diagnostics.</param>
+        /// <param name="diagnostics"></param>
+        /// <returns></returns>
+        private BoundPatternSwitchSection BindPatternSwitchSection(
+            SwitchSectionSyntax node,
+            Binder originalBinder,
+            ref BoundPatternSwitchLabel defaultLabel,
+            ref bool someCaseMatches,
+            SubsumptionDiagnosticBuilder subsumption,
+            DiagnosticBag diagnostics)
+        {
+            // Bind match section labels
+            var boundLabelsBuilder = ArrayBuilder<BoundPatternSwitchLabel>.GetInstance();
+            var sectionBinder = originalBinder.GetBinder(node); // this binder can bind pattern variables from the section.
+            Debug.Assert(sectionBinder != null);
+            var labelsByNode = LabelsByNode;
+
+            foreach (var labelSyntax in node.Labels)
+            {
+                LabelSymbol label = labelsByNode[labelSyntax];
+                BoundPatternSwitchLabel boundLabel = BindPatternSwitchSectionLabel(sectionBinder, labelSyntax, label, ref defaultLabel, diagnostics);
+                boundLabelsBuilder.Add(boundLabel);
+            }
+
+            // Bind switch section statements
+            var boundStatementsBuilder = ArrayBuilder<BoundStatement>.GetInstance();
```

Comment:
>GetInstance [](start = 70, length = 11)

Can we predict the size for the builder?

Reply:
Resolved in [https://github.com/dotnet/roslyn/pull/23357/commits/758fd76932a6d9bcb640504c8fdef57fb5adecec](https://github.com/dotnet/roslyn/pull/23357/commits/758fd76932a6d9bcb640504c8fdef57fb5adecec) of [https://github.com/dotnet/roslyn/pull/23357](https://github.com/dotnet/roslyn/pull/23357)

---
In reply to: [156525958](https://github.com/dotnet/roslyn/pull/23209#discussion_r156525958) [](ancestors = 156525958)